### PR TITLE
add family and youth vispdats

### PR DESCRIPTION
**Pull request by peterclark**
@eanders - this is just for the youth vispdat. I still need to add some tests but wanted to get this to you for any feedback. I'm still working on the family vispdat.

### CODE REVIEW

File: app/views/window/clients/vispdats/_list.haml [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
-      = link_to polymorphic_path(vispdats_path_generator, client_id: @client.id), class: 'btn btn-primary', title: 'New', data: { method: :post } do
-        %span.icon-plus
-        New VI-SPDAT
+      .btn-group
+        %button.btn.btn-default.dropdown-toggle(type='button' data-toggle='dropdown')
+          New VI-SPDAT
+          %span.caret
+        %ul.dropdown-menu
+          %li
+            = link_to polymorphic_path(vispdats_path_generator, client_id: @client.id, type: 'GrdaWarehouse::Vispdat::Youth'), data: { method: :post } do
```
Comment (eanders) [inline app/views/window/clients/vispdats/_list.haml]: We don't generally use font-awesome in this project, I'll see about adding them via the icomoon setup.

File: app/views/window/clients/vispdats/_vispdat.haml [hunk 2]
```diff
@@ -146,6 +146,12 @@
       %label Do you have any legal stuff going on right now that may result in you being locked up, having to pay fines, or that make it more difficult to rent a place to live?
     .col-sm-3
       %dd= yes_no @vispdat.answer_for :legal_answer
+  - if @vispdat.is_a? GrdaWarehouse::Vispdat::Youth
```
Comment (eanders) [inline app/views/window/clients/vispdats/_vispdat.haml]: Let's make  `youth?`, `family?`, and `individual?` methods on the models.

**Comment by peterclark** [comment]
@eanders - ok family was tricky, the auto update feature did not play well with the children nested attributes, but found a solution which we can discuss after I demo. Still need to add tests for family and youth vispdats.

**Comment by peterclark** [comment]
@eanders - this now includes the specific wording for family vispdats.

File: app/controllers/clients/vispdats_controller.rb [hunk 3]
```diff
@@ -5,6 +5,7 @@ class VispdatsController < Window::Clients::VispdatsController
     before_action :require_can_edit_vspdat!
 
     def destroy
+      @vispdat.disassociate_files
```
Comment (eanders) [inline app/controllers/clients/vispdats_controller.rb]: Any reason not to put this in a before_destroy?

File: app/views/window/clients/vispdats/_list.haml (hunk 1 above)
Comment (eanders) [reply app/views/window/clients/vispdats/_list.haml]: Check `/style_guide/icon_font` for icomoon versions of these.  You should be able to use `.icon-group` `.icon-individual` and `.icon-child` I believe.

File: app/controllers/clients/vispdats_controller.rb (hunk 3 above)
Comment (peterclark) [reply app/controllers/clients/vispdats_controller.rb]: I initially tried that, but something about when the foreign key association is checked prevented it from working.

**Comment by peterclark** [comment]
@eanders - tests have been added for scoring.

**Comment by peterclark** [comment]
@eanders - believe this is good to go now, addressed your comments and added tests to verify scoring.

**Pull request merged**